### DISCUSSION TITLE
ESPHome device (NSPanel) does not show up in Blueprint configuration page.

### DISCUSSION THREAD

**Discussion by st2000** [Blackymas/NSPanel_HA_Blueprint#2208] [Q&A]
Hi,

Background: I started out on an RPi3B and migrated to an RPi4 ... in case that makes a difference.

This is the 1st ESPHome device I have tried to bring up. The NSPanel formerly had Tasmota running on it and HA did work well with it. I then programmed the NSPanel with ESPHome and it has been a battle to get things to work.  I have gotten to the point where I am trying to associate the Blueprint with this device.  But the drop down menu lists no ESPHome devices to choose from.

Side note: The ESPHome device does show up in the HA ESPHome app.  And in the 1 and only box (recall I only have the 1 ESPHome device) it does say "on line".

This is confusing as, perhaps hours earlier, I did see the NSPanel ESPHome device name on this drop down.  But not now.  Further, there were many instances of Blueprint listed in HA that appeared to be associated with the NSPanel.  To further confuse things, I could not delete these incorrect / undesired instance of Blueprints. In the end, I managed to delete all but 1 Blueprint.  (Sorry for how vague this bit of the question is.  If I find out more I'll come back and edit it.)

Also, I am a bit worried that I have never seen a "check-box" for uploading a TFT file.  It could be that I will not see any mention of a TFT file until this Blueprint mess is cleared up.  But still, I keep reading about how to upload a TFT file for the NSPanel display but never see where to do it on my instance of HA.

-thanks

**Comment by edwardtfn** [comment]
When you install an ESPHome device, if the API is set (and this is the case in this project), that device should show up as a new device in your integrations page (Settings > Devices and Services) and then added to Home Assistant, and this is independent of being visible under the ESPHome Dashboard (from the add-on).

1. Do you have an ESPHome integration under Settings > Devices and Services? Is your panel listed there?
2. Do you have any special settings in your network? VLANs, etc. could block mDNS used by ESPHome.

**Reply by st2000** [reply]
> Do you have an ESPHome integration under Settings > Devices and Services? Is your panel listed there?

I believe I do. This is the page where every service (i.e. ESPHome or Tasmota or ...) has a box.  And in that box there are a number of devices? I recall there being 1 ESPHome device. (But I will double check when next I have access to my instance of HA.)

> Do you have any special settings in your network? VLANs, etc. could block mDNS used by ESPHome.

No, should just be an ordinary network. No VLAN. That said, I do depend on the DHCP server to serve up the same IP addresses as I tend to use IP numbers rather than names. But I'm usually looking-at / using-the numbers and I think they have remained static. That said, recall I migrated to a different HW platform (RPi3B -> RPi4). That caused the HA IP number to change. But, in defense of that change, the HueHub devices in HA continued to work.  And the Tasmota devices in HA started working as soon as I reinstalled the HA Tasmota Add-Ons.

Do you think the migration broke the ESPHome device?  Maybe I should delete it and add it again?

**Reply by edwardtfn** [reply]
Dynamic IP shouldn't be an issue, and I don't think the migration would be an issue also.
It would be nice if you can double check if that esphome device is shown there. The upload TFT option will be there.

**Reply by st2000** [reply]
>> Do you have an ESPHome integration under Settings > Devices and Services? Is your panel listed there?

> I believe I do. This is the page where every service (i.e. ESPHome or Tasmota or ...) has a box. And in that box there are a number of devices? I recall there being 1 ESPHome device. (But I will double check when next I have access to my instance of HA.)

Yes. Under Settings -> Devices & services there is a box for ESPHome and there is "1 DEVICE" listed in that box.

Clicking on "ESPHome" takes me to a page where it lists "nspanel100".  I can click "CONFIGURE" there to allow this device to be accessed by HA. Then it says "Success". No where on this page do I see a way to upload TFT files.

This is interesting! If I click Settings -> Devices & services -> (in the ESPHome box) 1 DEVICES it says in the Device info box "FrontDoor by Tasmota"! Hum, I thought I had deleted Tasmota. Maybe there is some place in HA that still needs to be "cleaned out"?!

----
Added later...

To combat the "Tasmota/NSPanel" problem, I deleted ESPHome and re-added it. That appears to have gotten rid of the "Tasmota/NSPanel" problem. But, when I click on Settings -> Devices & services -> ESPHome box I now see 1 device (that's good) but 38 entities!!!! What the heck?! This can't be good, can it??

When I click on entities, I see all kinds of entries.  From "Baud rate" to "nspanel100 Blueprint" to "nspanel100 Restart" to "Update TFT display - Baud rate".  Should I start deleting all of these??  What I was expecting was the 1 entry for "nspanel100 Blueprint". 

There is still no signs of a TFT UpLoad button or checkbox or dropdown.  Indicating, I assume, there is still something not configured correctly.

I'm not sure what to expect at this point.  Assuming more "things" are correctly configure, I'm going to try to run "Install nspanel100.yaml" again.   And see if that fixes things like missing TFT upload buttons.

----
Added later...

The web page at Settings -> Devices & services -> ESPHome box -> 1 devices is getting populated.  There are several new offerings which can be added to dashboards. There is one called Configuration.  In that box it appears to have some TFT  drop downs and buttons.
